Bottom: 4389e5ccd819a6d0803f1527aeca1074994ff4ce
Top:    3dc2d85df261635ce775f5736dbd30738699fc9a
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2013-07-18 13:26:15 -0700

rsockets: Support native IB addressing on connected rsockets

Update rsockets to support AF_IB addresses on connected rsockets.
Support for datagram rsockets is more difficult as a result of
using real UDP sockets for QP resolution, so that support is
deferred.  For connected sockets, we need to update internal
checks to handle AF_IB.

Allow the user to specify GID addresses into the rsocket sample
applications.

Signed-off-by: Sean Hefty <sean.hefty@intel.com>


---

diff --git a/src/rsocket.c b/src/rsocket.c
index abdd392..1a6d989 100644
--- a/src/rsocket.c
+++ b/src/rsocket.c
@@ -1063,7 +1063,7 @@ int rsocket(int domain, int type, int protocol)
 	struct rsocket *rs;
 	int index, ret;
 
-	if ((domain != PF_INET && domain != PF_INET6) ||
+	if ((domain != AF_INET && domain != AF_INET6 && domain != AF_IB) ||
 	    ((type != SOCK_STREAM) && (type != SOCK_DGRAM)) ||
 	    (type == SOCK_STREAM && protocol && protocol != IPPROTO_TCP) ||
 	    (type == SOCK_DGRAM && protocol && protocol != IPPROTO_UDP))
